src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -390,22 +400,24 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
     periodObj[bufferType].dispatcher = dispatcher;
 
     dispatcher.addEventListener("noPlayableRepresentation", () => {
-      const nextAdaptation = arrayFind(
-        period.adaptations[bufferType] ?? [],
-        (adaptation) => {
-          if (
-            adaptation.supportStatus.hasSupportedCodec === false ||
-            adaptation.supportStatus.isDecipherable === false
-          ) {
-            return false;
-          }
-          const playableRepresentations = adaptation.representations.filter(
-            (r) => isRepresentationPlayable(r) === true,
-          );
-          return playableRepresentations.length > 0;
-        },
-      );
-      if (nextAdaptation === undefined) {
+      const nextAdaptation = findNextPlayableAdaptation(period, bufferType);
+      if (
+        nextAdaptation === undefined &&
+        bufferType === "audio" &&
+        this.onAudioTracksNotPlayable === "continue" &&
+        findNextPlayableAdaptation(period, "video") !== undefined
```

Comment:
Why are we doing this call? We just have to check that the current video track is defined no?

Also what happens if we were in audio-only mode here?

Reply:
I have updated the code since, the comment seems not relevant anymore